src/app/InteractionModelEngine.h

```diff
@@ -306,6 +306,9 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
     CHIP_ERROR ResumeSubscriptions();
 
+    // Return the state of a subscription matching a CAT Tag or a NodeId
+    bool IsMatchingSubscriptionActive(const FabricIndex & aFabricIndex, const NodeId & nodeIdOrCAT);
```

Comment:
naming isn't clear with what the function does; we aren't matching.
maybe `doesSubjectHaveActiveSubscription`? or a verison that with the associated comment.

Reply:
Well to me we are indeed Matching the subscription against the fabricId AND (NodeId OR CAT). 

What would be a Match if not the above ?

Will make the comment/function name clearer